Using Windows Subsystem Linux  for Windows build will fail due to -fPIC

Following the steps to build a windows version with Windows Subsystem Linux installed causes -fPIC errors

clang: fatal error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'

Main issue is when you type 'bash' it starts up and instance of WSL.  When running configure this logic is applied

```
        os_name=$(uname -s)
	os_vers=$(uname -r)
	echo "${script_name}: detected ${os_name} kernel version ${os_vers}."

	# Define a single variable off of which we can branch to tell if we are
	# building for Windows.
	is_win=no
	if [[ $os_name == MSYS* ]] || \
	   [[ $os_name == MINGW* ]]  || \
	   [[ $os_name == CYGWIN* ]] ; then
		is_win=yes
	fi
```

Thus is_win is not set to true,   as when using WSL "uname -s"  it returns "Linux" !!!  (as it should as it is running linux)
Apart from manually modifying this not sure how this should be fixed.   
A user with WSL installed could be using it to build a proper Linux version or a Windows version, but using "os_name" will not indicate what the user intended!!

Yes, it works. There are more errors with that cross compiling build though.
1. Needs `-fno-asynchronous-unwind-tables` (we should check `__MINGW64__` macro and add that flag)
2. Next error is,

```
x86_64-w64-mingw32-gcc -O2 -O3 -fomit-frame-pointer -mavx2 -mfma -mfpmath=sse -march=haswell -fno-asynchronous-unwind-tables -Wall -Wno-unused-function -Wfatal-errors -std=c99 -D_POSIX_C_SOURCE=200112L -Iinclude/x86_64 -I./frame/3/ -I./frame/ind/ukernels/ -I./frame/3/ -I./frame/1m/ -I./frame/1f/ -I./frame/1/ -I./frame/include -DBLIS_VERSION_STRING=\"0.8.1-17\" -DBLIS_IS_BUILDING_LIBRARY -Wl,--exclude-all-symbols -c kernels/haswell/3/sup/bli_gemmsup_rv_haswell_asm_d6x8m.c -o obj/x86_64/kernels/haswell/3/sup/bli_gemmsup_rv_haswell_asm_d6x8m.o
kernels/haswell/3/sup/bli_gemmsup_rv_haswell_asm_d6x8m.c: In function ‘bli_dgemmsup_rv_haswell_asm_6x6m’:
kernels/haswell/3/sup/bli_gemmsup_rv_haswell_asm_d6x8m.c:1763:1: error: bp cannot be used in asm here
 }
 ^
compilation terminated due to -Wfatal-errors.
Makefile:601: recipe for target 'obj/x86_64/kernels/haswell/3/sup/bli_gemmsup_rv_haswell_asm_d6x8m.o' failed
make: *** [obj/x86_64/kernels/haswell/3/sup/bli_gemmsup_rv_haswell_asm_d6x8m.o] Error 1
```

--- a/configure
+++ b/configure
@@ -2373,6 +2373,11 @@ main()
 	fi
 
 	echo "${script_name}: using '${found_cc}' C compiler."
+	
+	# Also check the compiler to see if we are (cross-)compiling for Windows
+	if ${found_cc} -dM -E - < /dev/null 2> /dev/null | grep -q _WIN32; then
+		is_win=yes
+	fi
 
 
 	# -- Find a C++ compiler ---------------------------------------------------

